File: src/main/java/com/countryinfoscraper/CountryParser.java
package com.countryinfoscraper;

import org.jsoup.nodes.Document;

public class CountryParser {

    /**
     * Orchestrates the parsing of a Wikipedia country document.
     * Delegates specific parsing tasks to specialized parsers.
     */
    public static Country parseCountry(Document doc) {
        Country country = new Country();
        
        // Use InfoboxParser for structured data table
        InfoboxParser.parse(doc, country);
        
        // Use DescriptionParser for the summary paragraph
        DescriptionParser.parse(doc, country);

        return country;
    }
}

File: src/parsers/country-parser.ts
import { CheerioAPI } from 'crawlee';
import { Country } from '../types/country.js';
import { InfoboxParser } from './infobox.js';
import { DescriptionParser } from './description.js';

export class CountryParser {
  static parseCountry($: CheerioAPI): Partial<Country> {
    const country: Partial<Country> = {};
    InfoboxParser.parse($, country);
    DescriptionParser.parse($, country);
    return country;
  }
}
